sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -836,10 +836,6 @@ private async Task RunReceiveTaskAsync(CancellationToken cancellationToken)
                         {
                             await _messageHandlerSemaphore.WaitAsync(linkedHandlerTcs.Token).ConfigureAwait(false);
                             messageHandlerLockAcquired = true;
-                            if (IsSessionProcessor)
```

Comment:
This was causing us to wait unnecessarily when MaxCallsPerSession > 1 but there were outstanding accept session calls in progress.